Added periodic job for building kueue image for s390x and power

Deepali1999 opened the pull request:
## PR Description
Added periodic jobs to build and test Kueue on s390x and ppc64le architectures. This helps ensure Kueue works properly on these platforms.

### Added files
- `kueue-periodic-s390x-ppc64le-main.yaml`: Periodic job for main branch
- `kueue-periodics-s390x-ppc64le-release-0.11.yaml`: Periodic job for release-0.11
- `kueue-periodics-s390x-ppc64le-release-0.12.yaml`: Periodic job for release-0.12

### Changes include
- Added 12h interval periodic jobs for unit tests
- Configured for both s390x and ppc64le platforms
- Using kubekins-e2e image with Docker-in-Docker support

### Testing Done
- Tested locally.
- Image build is working.

kannon92 commented on config/jobs/kubernetes-sigs/kueue/kueue-periodic-s390x-ppc64le-main.yaml:
```diff
@@ -0,0 +1,48 @@
+        - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20250527-1b2b10e804-master
+          env:
+            - name: E2E_K8S_VERSION
+              value: "1.30"
```
This k8s is out of support.

If you look at the kueue periodics, you will see that we typically have e2e test on 3 versions of Kubernetes (1.33, 1.32, 1.31). I would suggest to follow that pattern here.

Deepali1999 replied:
> This k8s is out of support.
> 
> If you look at the kueue periodics, you will see that we typically have e2e test on 3 versions of Kubernetes (1.33, 1.32, 1.31). I would suggest to follow that pattern here.

Thanks Kevin for the review I have update the Kubernetes version from 1.30 to 1.33 in below jobs.Please let me know if any further modifications are required.

kueue-periodic-s390x-ppc64le-main.yaml,  kueue-periodics-s390x-ppc64le-release-0.11.yaml,  kueue-periodics-s390x-ppc64le-release-0.12.yaml

kannon92 reviewed (changes requested):
The name should reflect the job. I also don’t know what value having a periodic job for building the image has..

id be more interested in knowing if this image works e2e?

Deepali1999 commented:
> The name should reflect the job. I also don’t know what value having a periodic job for building the image has..
> 
> id be more interested in knowing if this image works e2e?

> The name should reflect the job. I also don't know what value having a periodic job for building the image has..
> 
> id be more interested in knowing if this image works e2e?

Thanks @kannon92!. I've made the following updates to address your concerns:

### Changes Made:

1. **Fixed Job Names**: Updated all job names to reflect what they actually do:
   - `periodic-kueue-test-unit-s390x-ppc64le-main` → `periodic-kueue-build-s390x-ppc64le-main`
   - `periodic-kueue-test-unit-s390x-ppc64le-release-0.11` → `periodic-kueue-build-s390x-ppc64le-release-0.11`
   - `periodic-kueue-test-unit-s390x-ppc64le-release-0.12` → `periodic-kueue-build-s390x-ppc64le-release-0.12`

2. **Updated Descriptions**: Changed descriptions to reflect the actual functionality:
   - "Run s390x periodic kueue unit tests" → "Build kueue images on s390x and ppc64le platforms"

The jobs now build the images & test them.
/cc @kannon92

mimowo commented:
@Deepali1999 thank you for the PR. I think you need to adjust it for 0.12 and 0.13 as these are the currently supported versions.

Deepali1999 commented:
> @Deepali1999 thank you for the PR. I think you need to adjust it for 0.12 and 0.13 as these are the currently supported versions.

Thanks @mimowo, I have updated to add support for  release 0.12 and release 0.13.

mimowo commented:
> /hold
> 
> https://github.com/kubernetes/test-infra/pull/35175/files#r2231260185

/unhold

This is already addressed as 1.33 is now used.

mimowo commented:
@Deepali1999 there are failures actually, https://github.com/kubernetes-sigs/kueue/issues/6517 ptal.

mbobrovskyi commented on config/jobs/kubernetes-sigs/kueue/kueue-periodic-s390x-ppc64le-main.yaml:
```diff
@@ -0,0 +1,47 @@
+            # generic runner script, handles DIND, bazelrc for caching, etc.
+            - runner.sh
+          args:
+            - make
+            - image-build
+            - PLATFORMS=linux/s390x,linux/ppc64le
```
Why we have this test? This is just build. Do we need to test it periodically?

mimowo replied:
IIUC the intention is to eventually offload the presubmit builds. However, we need to first prepare also the release builds for that, so we may temporarily revert.

see https://github.com/kubernetes-sigs/kueue/issues/5166#issuecomment-2865199715

mbobrovskyi replied:
I see. Thank you for explanation.

Deepali1999 replied:
Thanks @mbobrovskyi I have raised another PR to remove the support for the power https://github.com/kubernetes/test-infra/pull/35296 until the multi arch builds are supported.
